mage_ai/data_preparation/repo_manager.py

```diff
@@ -41,30 +76,15 @@ def __init__(self, repo_path: str = None, config_dict: Dict = None):
             else:
                 repo_config = config_dict
 
-            # Priority:
-            # 1. 'variables_dir' from config_dict
-            # 1. os.getenv(MAGE_DATA_DIR_ENV_VAR)
-            # 2. 'variables_dir' from project's metadata.yaml file
-            # 3. DEFAULT_MAGE_DATA_DIR
-            if config_dict is not None and config_dict.get('variables_dir'):
-                self.variables_dir = config_dict.get('variables_dir')
-            elif os.getenv(MAGE_DATA_DIR_ENV_VAR):
-                self.variables_dir = os.getenv(MAGE_DATA_DIR_ENV_VAR)
+            if config_dict and config_dict.get('variables_dir'):
```

Comment:
why don't you put this inside the `get_variables_dir` method?

Reply:
because this seems like a different flow (where the config_dict is directly passed into the `RepoConfig` object). It seems separate from fetching the repo config from the project `metadata.yaml` file.

Reply:
but there's duplicate logic in the `get_variables_dir`
```
    elif repo_config is not None:
        variables_dir = repo_config.get('variables_dir')
```

Reply:
oh that's supposed to be different. The `repo_config` in `get_variables_dir` is the config from the project `metadata.yaml`. 

I only added the `repo_config` parameter so that if the caller method already fetched the project `metadata.yaml`, it could just pass it in to `get_variables_dir` so that it doesn't try to read from the `metadata.yaml` again.